pkg/payerreport/workers/attestation.go

```diff
@@ -0,0 +1,180 @@
+package workers
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/xmtp/xmtpd/pkg/payerreport"
+	"github.com/xmtp/xmtpd/pkg/registrant"
+	"github.com/xmtp/xmtpd/pkg/tracing"
+	"go.uber.org/zap"
+)
+
+// attestationWorker is responsible for periodically checking for reports that need attestation
+// and signing them with the node's private key.
+type attestationWorker struct {
+	ctx          context.Context
+	cancel       context.CancelFunc
+	log          *zap.Logger
+	registrant   registrant.IRegistrant
+	store        payerreport.IPayerReportStore
+	verifier     payerreport.PayerReportVerifier
+	wg           sync.WaitGroup
+	pollInterval time.Duration
+}
+
+// NewAttestationWorker creates and starts a new attestation worker that will periodically
+// check for reports that need attestation.
+// It takes a context, logger, registrant for signing, store for accessing reports,
+// and a poll interval that determines how often to check for reports.
+func NewAttestationWorker(
```

Comment:
does not seem to be hooked up yet?

Reply:
That's right. Need a few more pieces in place before I start running the worker.